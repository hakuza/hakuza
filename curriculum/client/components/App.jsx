import React from 'react';
import axios from 'axios';
import ChapterList from './ChapterList.jsx';

export default class App extends React.Component {
  constructor(props) {
    super(props);

  }

  // onClick(e) {   e.preventDefault();   $('.expand-one').click(() => {
  // console.log('clicked');     $('.content-one').slideToggle('slow');   }); }

  render() {
    return (
      <div id="container">
        <h2>Curriculum For This Course</h2>
        <h5>Expand All</h5>
        <h5>46 Lectures</h5>
        <h5>05:04:18</h5>
        <ChapterList/>
      </div>
    );
  }
}
