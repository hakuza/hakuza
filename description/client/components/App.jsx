import React from 'react';
import axios from 'axios';
import Feedback from './Feedback.jsx';

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      course: [],
      currentCourse: {}
    };
  }

  componentWillMount() {
    axios
      .get('/description')
      .then(response => {
        this.setState({
          course: response.data,
          currentCourse: response.data[16]
        });
      })
      .catch(err => console.log(err));
  }

  render() {
    return (
      <div>
        <h1>{this.state.currentCourse.title}</h1>
        <h2>{this.state.currentCourse.headline}</h2>
        <Feedback currentCourse={this.state.currentCourse} />
        <h4>Created by {this.state.currentCourse.name}</h4>
        <h4>
          Last Updated{' '}
          {new Date(this.state.currentCourse.created).toDateString()}
        </h4>
        <h4>English</h4>
        <h4>Closed captioning</h4>
      </div>
    );
  }
}
