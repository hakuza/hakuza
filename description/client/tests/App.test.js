import { shallow, mount } from 'enzyme';
import React from 'react';
import App from '../components/App.jsx';
import React from 'react';
import ReactDOM from 'react-dom';
import App from './components/App.jsx';

ReactDOM.render(<App />, document.getElementById('description'));

it('works and doesnt crash', () => {
  const wrap = mount(<App />);
  expect(wrap).toMatchSnapshot();
});
