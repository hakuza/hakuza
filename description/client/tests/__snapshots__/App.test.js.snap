// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
<App>
  <div>
    <h1 />
    <h2 />
    <Feedback
      currentCourse={Object {}}
    >
      <div>
        <h4>
          histogram
        </h4>
        <h4 />
        <h4>
          (
           ratings)
        </h4>
        <h4>
           students enrolled
        </h4>
      </div>
    </Feedback>
    <h4>
      Created by 
    </h4>
    <h4>
      Last Updated
       
      Invalid Date
    </h4>
    <h4>
      English
    </h4>
    <h4>
      Closed captioning
    </h4>
  </div>
</App>
`;
